tests/cross/src/test/scala/tests/pc/TypeDefinitionSuite.scala

```diff
@@ -0,0 +1,42 @@
+package tests.pc
+
+object TypeDefinitionSuite extends BaseTypeDefinitionSuite {
+
+  check(
+    "basic",
+    """
+      |class Main(i: Int) {
+      |
+      |}
+      |
+      |object Main {
+      |  val ts@@t = new Main(2)
+      |}""".stripMargin,
+    "Main.scala/\\2:6/\\2:9"
+  )
+
+  check(
+    "method",
+    """
+      |class Main(i: Int) {}
+      |
+      |object Main {
+      | def tst(m: Main): Unit = {}
+      |
+      |  tst(new M@@ain(2))
+      |}""".stripMargin,
+    "Main.scala/\\2:6/\\2:9"
+  )
+
+  check(
```

Comment:
Maybe add tests additionaly for:
-  named parameter -> it should go to the parameter inside the method
-  on pattern match - it should go to corresponding unapply if it exists otherwise to the object

Anything more you can come up with while testing.